include/RevMem.h

```diff
@@ -313,25 +325,30 @@ class RevMem{
   char *physMem = nullptr;                 ///< RevMem: memory container
 
 private:
+  unsigned long memSize;        ///< RevMem: size of the target memory
+  unsigned tlbSize;             ///< RevMem: number of entries in the TLB
+  unsigned maxHeapSize;         ///< RevMem: maximum size of the heap
   std::unordered_map<uint64_t, std::pair<uint64_t, std::list<uint64_t>::iterator>> TLB;
   std::list<uint64_t> LRUQueue; ///< RevMem: List ordered by last access for implementing LRU policy when TLB fills up
-  std::vector<std::shared_ptr<MemSegment>> MemSegs;     // Currently Allocated MemSegs
-  std::vector<std::shared_ptr<MemSegment>> FreeMemSegs; // MemSegs that have been unallocated
-  unsigned long memSize;        ///< RevMem: size of the target memory
-  unsigned tlbSize;             ///< RevMem: size of the target memory
-  unsigned maxHeapSize;             ///< RevMem: size of the target memory
-
   RevOpts *opts;                ///< RevMem: options object
   RevMemCtrl *ctrl;             ///< RevMem: memory controller object
   SST::Output *output;          ///< RevMem: output handler
 
+
+  std::vector<std::shared_ptr<MemSegment>> MemSegs;       // Currently Allocated MemSegs
+  std::vector<std::shared_ptr<MemSegment>> FreeMemSegs;   // MemSegs that have been unallocated
+  std::vector<std::shared_ptr<MemSegment>> ThreadMemSegs; // For each RevThread there is a corresponding MemSeg that contains TLS & Stack
+
+  uint64_t TLSBaseAddr;                                   ///< RevMem: TLS Base Address
+  uint64_t TLSSize = sizeof(uint32_t);                    ///< RevMem: TLS Size (minimum size is enough to write the TID)
+  uint64_t ThreadMemSize = _STACK_SIZE_;                  ///< RevMem: Size of a thread's memory segment (StackSize + TLSSize)
+  uint64_t NextThreadMemAddr = memSize-1024;                  ///< RevMem: Next top address for a new thread's memory (starts at the point the 1024 bytes for argc/argv ends)
+
   uint64_t SearchTLB(uint64_t vAddr);                       ///< RevMem: Used to check the TLB for an entry
   void AddToTLB(uint64_t vAddr, uint64_t physAddr);         ///< RevMem: Used to add a new entry to TLB & LRUQueue
   void FlushTLB();                                          ///< RevMem: Used to flush the TLB & LRUQueue
   uint64_t CalcPhysAddr(uint64_t pageNum, uint64_t vAddr);  ///< RevMem: Used to calculate the physical address based on virtual address
-  bool isValidVirtAddr(const uint64_t vAddr);               ///< RevMem: Used to check if a virtual address exists in MemSegs
-
-  uint32_t PIDCount = 1023;   ///< RevMem: Monotonically increasing PID counter for assigning new PIDs without conflicts
+  bool isValidVirtAddr(const uint64_t& vAddr);               ///< RevMem: Used to check if a virtual address exists in MemSegs
```

Comment:
Passing `vAddr` by value should be faster than passing by reference, because it is a trivial scalar type `uint64_t`, and references always create aliasing opportunities the compiler might not optimize.